### PR TITLE
[#101020636] Fixes null weight in animated model

### PR DESCRIPTION
**Pull request by gfaucill**

### CODE REVIEW

File: sources/osgShader/node/skeletal.glsl [hunk 1]
```diff
@@ -14,6 +14,8 @@ mat4 skeletalTransform( const in vec4 weightsVec, const in vec4 bonesIdx ) {
     mat4 tmpMat = mat4(1.0);
     mat4 outMat = mat4(0.0);
 
+    if((weightsVec.x + weightsVec.y + weightsVec.z + weightsVec.w) == 0.0) return tmpMat;
```
Comment (cedricpinson) [inline sources/osgShader/node/skeletal.glsl]: @Kuranes do you have a trick to do this in an optimized way ?

Comment (Kuranes) [inline sources/osgShader/node/skeletal.glsl]: glsl vector comparison equal( vec, veczero) ?

Comment (cedricpinson) [inline sources/osgShader/node/skeletal.glsl]: @gfaucill tu peux faire l'update

**Comment by gfaucill** [comment]
@cedricpinson Done

File: sources/osgShader/node/skeletal.glsl [hunk 2]
```diff
@@ -14,6 +14,8 @@ mat4 skeletalTransform( const in vec4 weightsVec, const in vec4 bonesIdx ) {
     mat4 tmpMat = mat4(1.0);
     mat4 outMat = mat4(0.0);
 
+    if(all(equal(weightsVec, vec4(0.0)))) return tmpMat;
```
Comment (cedricpinson) [inline sources/osgShader/node/skeletal.glsl]: why all ? if it's not equal it should enough no ?

**Comment by gfaucill** [comment]
equal returns a vec4 of boolean != boolean

**Pull request merged**